addon/validations/factory.js

```diff
@@ -102,11 +104,15 @@ function setOwner(obj, model) {
 export default function buildValidations(validations = {}) {
   processDefaultOptions(validations);
 
-  const Validations = createValidationsObject(validations);
+  var Validations;
 
   return Ember.Mixin.create({
     validations: computed(function() {
-      return Validations.create({ model: this, _parentValidations: this._super() });
+      if(!Validations) {
+        // We only want to create this once per class
+        Validations = createValidationsClass(this._super(), validations);
```

Comment:
Makes sense to me. This will expose `model.validationsClass` as well. Maybe we should make it private. Ill give this a shot.